dm/dm/master/server_test.go

```diff
@@ -2359,3 +2378,132 @@ func (t *testMaster) validatorModeMatch(c *check.C, s *scheduler.Scheduler, task
+	server.etcdClient = t.etcdTestCli
+	sources, workers := defaultWorkerSource()
+	startReq := &pb.StartTaskRequest{
+		Task:    taskConfig,
+		Sources: sources,
+	}
+	// test query all workers
+	for idx, worker := range workers {
+		mockWorkerClient := pbmock.NewMockWorkerClient(ctrl)
+		mockWorkerClient.EXPECT().GetWorkerValidateStatus(
+			gomock.Any(),
+			gomock.Any(),
+		).Return(&pb.GetValidationStatusResponse{
+			Result: true,
+			Status: []*pb.ValidationStatus{
+				{
+					SrcTable: "tbl1",
+				},
+			},
+		}, nil)
+		mockRevelantWorkerClient(mockWorkerClient, taskName, sources[idx], startReq)
+		t.workerClients[worker] = newMockRPCClient(mockWorkerClient)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer t.clearSchedulerEnv(c, cancel, &wg)
+	// start task without validation
+	sourceResps := []*pb.CommonWorkerResponse{{Result: true, Source: sources[0]}, {Result: true, Source: sources[1]}}
+	server.scheduler, _ = t.testMockScheduler(ctx, &wg, c, sources, workers, "", t.workerClients)
+	mock := conn.InitVersionDB(c)
+	defer func() {
+		conn.DefaultDBProvider = &conn.DefaultDBProviderImpl{}
+	}()
+	mock.ExpectQuery("SHOW GLOBAL VARIABLES LIKE 'version'").WillReturnRows(sqlmock.NewRows([]string{"Variable_name", "Value"}).
+		AddRow("version", "5.7.25-TiDB-v4.0.2"))
+	stResp, err := server.StartTask(context.Background(), startReq)
+	c.Assert(err, check.IsNil)
+	c.Assert(stResp.Result, check.IsTrue)
+	for _, source := range sources {
+		t.subTaskStageMatch(c, server.scheduler, taskName, source, pb.Stage_Running)
+	}
+	c.Assert(stResp.Sources, check.DeepEquals, sourceResps)
+	// 1. query existing task's status
+	statusReq := &pb.GetValidationStatusRequest{
+		TaskName: taskName,
+	}
+	resp, err := server.GetValidationStatus(context.Background(), statusReq)
+	c.Assert(err, check.IsNil)
+	c.Assert(resp.Msg, check.Equals, "")
+	c.Assert(resp.Result, check.IsTrue)
+	c.Assert(len(resp.Status), check.Equals, 2)
+	// 2. query invalid task's status
+	statusReq.TaskName = "invalid-task"
+	resp, err = server.GetValidationStatus(context.Background(), statusReq)
+	c.Assert(err, check.IsNil)
+	c.Assert(resp.Msg, check.Matches, ".*fail to get subtask config by task name.*")
+	c.Assert(resp.Result, check.IsFalse)
+	// 3. query invalid stage
+	statusReq.TaskName = taskName
+	statusReq.FilterStatus = "invalid-stage"
+	resp, err = server.GetValidationStatus(context.Background(), statusReq)
+	c.Assert(err, check.IsNil)
+	c.Assert(resp.Msg, check.Matches, ".*filtering stage should be either.*")
+	c.Assert(resp.Result, check.IsFalse)
+}
+
+func (t *testMaster) TestOneSourceValidatorStatus(c *check.C) {
+	var (
+		wg       sync.WaitGroup
+		taskName = "new-task"
+	)
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+	server := testDefaultMasterServer(c)
+	server.etcdClient = t.etcdTestCli
+	sources, workers := defaultWorkerSource()
+	sources = sources[:1]
+	workers = workers[:1]
+	startReq := &pb.StartTaskRequest{
+		Task:    taskCfg2,
```

Comment:
seems this taskCfg2 is not valid?, and you used a mock client, maybe just pass a empty string as task cfg?

Reply:
I think it oughts to be valid because the `server.StartTask` will check its validity before starting a task and thus empty string cannot pass the verification.

Reply:
But this test is removed anyway...